src/days-2-4-adv/adv.py

```diff
@@ -49,3 +59,20 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+quit = False
+
+while not quit:
+    print("\n" + player.currentRoom.name + "\n" +
+          "\n" + player.currentRoom.description + "\n" + "")
+    cmd = input("=> ")
+    if cmd == "q":
+        quit = True
+    elif cmd == "n":
+        player.currentRoom = player.currentRoom.n_to
+    elif cmd == "s":
+        player.currentRoom = player.currentRoom.s_to
+    elif cmd == "e":
+        player.currentRoom = player.currentRoom.e_to
+    elif cmd == "w":
+        player.currentRoom = player.currentRoom.w_to
```

Comment:
I think in here, you need to add conditionals that will print an error message if there is no valid n_to, s_to, etc. and then reprint the input prompt.

Reply:
FYI, this is still not working correctly.